pyquil/tests/test_quil.py

```diff
@@ -380,3 +380,9 @@ def test_extract_qubits():
     p.inst(X(new_qubit))
     p.synthesize()
     assert p.extract_qubits() == set([0, 1, 2, 3, 4, 5, 6, new_qubit.index()])
+
+
+def test_prog_merge():
+    prog_0 = Program(X(0))
+    prog_1 = Program(Y(0))
+    assert merge_programs([prog_0, prog_1]) == prog_0 + prog_1
```

Comment:
And also this is failing in the tests - they're different objects. It might make sense to overload  `__eq__`?